### ISSUE TITLE
Wrong name mangling. V8Utils => v8utils instead of v8.utils

### ISSUE THREAD

**Issue by Gonzih** [jruby/jruby#3493]
Hello, 

I found small issue with ruby => java package name mangling.
Here is an example:

```ruby
Java::ComEclipsesourceV8Utils::V8Executor
NameError: cannot load Java class com.eclipsesource.v8utils.V8Executor
from org/jruby/javasupport/JavaClass.java:204:in `for_name'
```

V8Executor is located inside `com.eclipsesource.v8.utils` package.

Thanks!

**Comment by headius** [comment]
I suspect there's just a bad regex looking for [a-z][A-Z] to split rather than including other valid package characters in the first part.

**Comment by Gonzih** [comment]
Also have similar problem that might be related:

```ruby
java_import com.eclipsesource.v8.V8
NameError: missing class name (`com.eclipsesource.v8.V8')
               get_proxy_or_package_under_package at org/jruby/javasupport/JavaUtilities.java:54
```
Is this related or is this a different issue entirely?

**Comment by headius** [comment]
Is that the correct class + package for that V8 class? The dotted form is what we recommend users use to access Java classes, and I don't see a reason this would fail.

**Comment by headius** [comment]
I have a trivial fix for the main issue:

```diff
diff --git a/core/src/main/java/org/jruby/javasupport/Java.java b/core/src/main/java/org/jruby/javasupport/Java.java
index 9ba8ba4..8108217 100644
--- a/core/src/main/java/org/jruby/javasupport/Java.java
+++ b/core/src/main/java/org/jruby/javasupport/Java.java
@@ -766,7 +766,7 @@ public class Java implements Library {
         return packageModule;
     }
 
-    private static final Pattern CAMEL_CASE_PACKAGE_SPLITTER = Pattern.compile("([a-z][0-9]*)([A-Z])");
+    private static final Pattern CAMEL_CASE_PACKAGE_SPLITTER = Pattern.compile("([a-z0-9]+)([A-Z])");
 
     private static RubyModule getPackageModule(final Ruby runtime, final String name) {
         final RubyModule javaModule = runtime.getJavaSupport().getJavaModule();
```

**Comment by headius** [comment]
What version of JRuby are you using? We could fix this in both 1.7 and 9 but we'd prefer to just fix it in 9 since it may change the way existing code resolves packages (even if they do happen to be using the WRONG way, we don't want to break people's code).

**Comment by Gonzih** [comment]
I'm using 9.0.4.0, thanks for the fix! I will look at the java import issue closer.
